lms/static/js/learner_dashboard/views/course_enroll_view_2017.js

```diff
@@ -0,0 +1,111 @@
+(function(define) {
+    'use strict';
+
+    define(['backbone',
+        'jquery',
+        'underscore',
+        'gettext',
+        'edx-ui-toolkit/js/utils/html-utils',
+        'text!../../../templates/learner_dashboard/course_enroll_2017.underscore'
+    ],
+         function(
+             Backbone,
+             $,
+             _,
+             gettext,
+             HtmlUtils,
+             pageTpl
+         ) {
+             return Backbone.View.extend({
+                 className: 'course-enroll-view',
+
+                 tpl: HtmlUtils.template(pageTpl),
+
+                 events: {
+                     'click .enroll-button': 'handleEnroll',
+                     'change .run-select': 'handleCourseRunSelect'
+                 },
+
+                 initialize: function(options) {
+                     this.$parentEl = options.$parentEl;
+                     this.enrollModel = options.enrollModel;
+                     this.urlModel = options.urlModel;
+                     this.render();
+                     if (this.urlModel) {
+                         this.trackSelectionUrl = this.urlModel.get('track_selection_url');
+                     }
```

Comment:
It seems like render should be called after all of the other code in the init function.